lib/framework/elastic_connector_actions.rb

```diff
@@ -12,88 +12,137 @@
+        Utility::Logger.info("Successfully updated configuration for connector #{connector_package_id}")
+      end
 
-    def self.claim_job(connector_package_id)
-      body = {
-        :doc => {
-          :sync_now => false,
-          :sync_status => Connectors::SyncStatus::IN_PROGRESS,
-          :last_synced => Time.now
+      def claim_job(connector_package_id)
+        body = {
+          :doc => {
+            :sync_now => false,
+            :sync_status => Connectors::SyncStatus::IN_PROGRESS,
+            :last_synced => Time.now
+          }
         }
-      }
 
-      client.update(:index => CONNECTORS_INDEX, :id => connector_package_id, :body => body)
-      Utility::Logger.info("Successfully claimed job for connector #{connector_package_id}")
-    end
+        client.update(:index => CONNECTORS_INDEX, :id => connector_package_id, :body => body)
+        Utility::Logger.info("Successfully claimed job for connector #{connector_package_id}")
+      end
 
-    def self.complete_sync(connector_package_id, error)
-      body = {
-        :doc => {
-          :sync_status => error.nil? ? Connectors::SyncStatus::COMPLETED : Connectors::SyncStatus::FAILED,
-          :sync_error => error,
-          :last_synced => Time.now
+      def complete_sync(connector_package_id, error)
+        body = {
+          :doc => {
+            :sync_status => error.nil? ? Connectors::SyncStatus::COMPLETED : Connectors::SyncStatus::FAILED,
+            :sync_error => error,
+            :last_synced => Time.now
+          }
         }
-      }
 
-      client.update(:index => CONNECTORS_INDEX, :id => connector_package_id, :body => body)
+        client.update(:index => CONNECTORS_INDEX, :id => connector_package_id, :body => body)
 
-      if error
-        Utility::Logger.info("Failed to sync for connector #{connector_package_id} with error #{error}")
-      else
-        Utility::Logger.info("Successfully synced for connector #{connector_package_id}")
+        if error
+          Utility::Logger.info("Failed to sync for connector #{connector_package_id} with error #{error}")
+        else
+          Utility::Logger.info("Successfully synced for connector #{connector_package_id}")
+        end
       end
-    end
 
-    def self.client
-      @client ||= Utility::EsClient.new
-    end
+      def client
+        @client ||= Utility::EsClient.new
+      end
 
-    # should only be used in CLI
-    def self.ensure_index_exists(index_name)
-      client.indices.create(:index => index_name) unless client.indices.exists?(:index => index_name)
-    end
+      # should only be used in CLI
+      def ensure_index_exists(index_name, body = {})
+        client.indices.create(:index => index_name, :body => body) unless client.indices.exists?(:index => index_name)
+      end
 
-    # should only be used in CLI
-    def self.ensure_connectors_index_exists
-      ensure_index_exists(CONNECTORS_INDEX)
+      def ensure_alias_exists(alias_name, body = {})
```

Comment:
What's up with the alias? Why do we have/need it?

Reply:
For migration purposes. See [here](https://docs.google.com/document/d/1_4VCIDhj-a3EVHIJ_tADm28AtqZdZvo4CW-vrYAWcIQ/edit#bookmark=id.iqzlyufomwsk)

Reply:
TIL, thanks for the explanation! In this case we might need to add logic to find out which version index we need to create based on what's the last known version, but probably could be done later

Reply:
If the alias/index does not exist, we can assume we need to create `_v1` index. 

I would like to keep this PR simple, as there can be a lot of edge cases, e.g. alias not existing but index existing. I added it as one of the topic to discuss in [Engineers sync](https://docs.google.com/document/d/1Z3t79RpKpML-4vsuFAavXzMqAcw_cEbDqoK7Ov33hVg/edit#)